src/olympia/translations/tests/test_commands.py

```diff
@@ -1,70 +1,114 @@
-        # description. The command should fix this one.
-        self.collection.description.update(
-            localized_string_clean='bar_collection_description',
-            modified=self.days_ago(42),
-        )
-        self.collection.description.reload()
+        # Create expected translations in 2 steps so that their autoid and id
+        # are set as they would in production: first through .new() to create
+        # the instance and "reserve" the id, then .save() to record everything
+        # and generate an autoid (pk).
+        self.translations = [
+            # Translation that shouldn't be touched.
+            Translation.new('<b>foo</b> bar', 'en-US'),
+            # PurifiedTranslation that shouldn't be touched.
+            PurifiedTranslation.new('<b>foo</b> bar', 'en-US'),
+            # PurifiedTranslation (allows HTML) that should be updated.
+            PurifiedTranslation.new(
+                '<b>foo</b> https://outgoing.prod.mozaws.net/v1/bar', 'en-US'
+            ),
+            # LinkifiedTranslation (allows only links) that should be updated.
+            LinkifiedTranslation.new(
+                '<b>foo</b> https://outgoing.prod.mozaws.net/v1/bar', 'en-US'
+            ),
+            # PurifiedTranslation (allows HTML) that should be updated but only
+            # when passing a custom outgoing url.
+            PurifiedTranslation.new(
+                '<b>meh</b> https://outgoing.stage.mozaws.net/v1/xyz', 'en-US'
+            ),
+            # LinkifiedTranslation (allows only links) that should be updated
+            # but only when passing a custom outgoing url.
+            LinkifiedTranslation.new(
+                '<b>meh</b> https://outgoing.stage.mozaws.net/v1/xyz', 'en-US'
+            ),
+            # PurifiedTranslation (allows HTML) that should be updated (url is
+            # present twice).
+            PurifiedTranslation.new(
+                '<b>foo2</b> https://outgoing.prod.mozaws.net/v1/bar '
+                'https://outgoing.prod.mozaws.net/v1/bar2',
+                'en-US',
+            ),
+            # LinkifiedTranslation (allows only links) that should be updated.
+            #  (url is present twice).
+            LinkifiedTranslation.new(
+                '<b>foo2</b> https://outgoing.prod.mozaws.net/v1/bar '
+                'https://outgoing.prod.mozaws.net/v1/bar2',
+                'en-US',
+            ),
+        ]
+        for translation in self.translations:
+            translation.save()
 
-    def test_queryset(self):
-        command = ProcessTranslationsCommand()
-        qs = command.get_base_queryset({})
-        pks = command.get_pks(
-            qs,
-            command.get_tasks()['reclean_collection_descriptions']['queryset_filters'],
-        )
-        assert list(pks) == [self.collection.description.pk]
+    @override_settings(
+        REDIRECT_URL='https://prod.outgoing.prod.webservices.mozgcp.net/v1/'
```

Comment:
should we be updating `REDIRECT_URL` in our settings files anyway? Or is that in a later issue?

Reply:
I almost did it - in dev/stage/prod we're already overriding it with custom values anyway, but figured it could wait until we have migrated everything.

Reply:
https://github.com/mozilla/addons-server/issues/19733